### ISSUE TITLE
Focus on input elements do not show keyboard on iOS

### ISSUE THREAD

**Issue by JStumpp** [ionic-team/capacitor#334]
Calling .focus() or setting autofocus does not bring up the keyboard on IOS. This is a know issue with WkWebView. @mlynch you merged a pull request on your cordova wkwebview plugin: https://github.com/ionic-team/cordova-plugin-wkwebview-engine/pull/171/files
I know, you do not want to use any hacks in capacitor, but what is the alternative. Calling focus and calling Keyboard.show()?

**Comment by JStumpp** [comment]
Adding this to the AppDelegate works. If you do not want to include it it might be a good thing for the docs?
```swift
     func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplicationLaunchOptionsKey: Any]?) -> Bool {
    setKeyboardRequiresUserInteraction(false)
    // Override point for customization after application launch.
    return true
  }

    typealias ClosureType =  @convention(c) (Any, Selector, UnsafeRawPointer, Bool, Bool, Any) -> Void
    func setKeyboardRequiresUserInteraction( _ value: Bool) {
        let sel: Selector = sel_getUid("_startAssistingNode:userIsInteracting:blurPreviousNode:userObject:")
        let WKContentView: AnyClass = NSClassFromString("WKContentView")!
        let method = class_getInstanceMethod(WKContentView, sel)
        let originalImp: IMP = method_getImplementation(method!)
        let original: ClosureType = unsafeBitCast(originalImp, to: ClosureType.self)
        let block : @convention(block) (Any, UnsafeRawPointer, Bool, Bool, Any) -> Void = {(me, arg0, arg1, arg2, arg3) in
            original(me, sel, arg0, !value, arg2, arg3)
        }
        let imp: IMP = imp_implementationWithBlock(block)
        method_setImplementation(method!, imp)
    }
```

**Comment by mlynch** [comment]
Thanks for following up. Any thoughts on the app store approvalness of that code?

**Comment by JStumpp** [comment]
I have not tested the approval. Could of course be an issue. I will give it a try as soon as we will publish our app with capacitor, stencil and ionicv4. But this will take a few month. I will close this for now.

**Comment by mlynch** [comment]
Where'd you find it? 🕵️

**Comment by JStumpp** [comment]
Here: https://medium.com/@harekarpranit/playing-with-wkwebviews-f2c4620ad47a

**Comment by mlynch** [comment]
I just found this: https://github.com/ionic-team/cordova-plugin-ionic-webview/issues/78

**Comment by JStumpp** [comment]
Thanks, that is bad. Such a pity that the WKWebview does not provide this necessary functionality.

**Comment by mlynch** [comment]
I haven't received other reports, so I'm not sure if that's the final word on this. Perhaps there's another way. Will investigate (going to reopen)

**Comment by mlynch** [comment]
Cool I slightly modified your code and it's in master. Will keep an eye on any issues with the app store. Thanks!

**Comment by JStumpp** [comment]
Wonderful, thanks!

**Comment by mlynch** [comment]
Heads up: the method selector above has been removed from iOS 11.3 betas, I patched a quick fix, but the feature won't work on iOS 11.3 until we find a workaround

**Comment by JStumpp** [comment]
@mlynch this could be the solution for 11.3: https://stackoverflow.com/a/48623286
Could you give it a try, I have not installed 11.3? Thanks!
